lambda/MAS/Services/StaticWebsiteService.cs

```diff
@@ -28,55 +30,87 @@ public class S3StaticWebsiteService : IStaticWebsiteService
+            foreach(var req in requests)
+                taskDict.Add(req, WriteFileAsync(req));
 
-            try
+            foreach(var task in taskDict)
             {
-                var response = await _amazonS3.PutObjectAsync(request);
-                return response.HttpStatusCode;
-            }
-            catch (Exception e)
-            {
-                _logger.LogError(e, $"Failed to write ${filePath} to S3: {e.Message}");
-                throw e;
+                responseCode = (await task.Value).HttpStatusCode;
+                if (responseCode != HttpStatusCode.OK)
+                {
+                    _logger.LogError($"Writing {task.Key} resulted in a status code of {responseCode}");
+                    return responseCode;
+                }
             }
+
+            if (_environmentConfig.Name != "local" && requests.Count() > 0)
+                responseCode = InvalidateCacheAsync(taskDict.Select(x => "/" + x.Key.FilePath).ToList());
+
+            return responseCode;
+
         }
-        public async Task<HttpStatusCode> WriteFileAsync(string filePath, Stream inputStream)
+        
+        private Task<PutObjectResponse> WriteFileAsync(StaticContentRequest item)
         {
             PutObjectRequest request = new PutObjectRequest()
             {
                 BucketName = _awsConfig.BucketName,
-                Key = filePath,
-                InputStream = inputStream
+                Key = item.FilePath
             };
 
-            try
-            {
-                var response = await _amazonS3.PutObjectAsync(request);
-                return response.HttpStatusCode;
-            }
-            catch (Exception e)
+            if (string.IsNullOrEmpty(request.ContentBody))
+                request.InputStream = item.ContentStream;
+            else
+                request.ContentBody = item.ContentBody;
+
+            return _amazonS3.PutObjectAsync(request);
+        }
+
+        private HttpStatusCode InvalidateCacheAsync(List<string> paths)
+        {           
+            var invalidationBatch = new InvalidationBatch()
             {
-                _logger.LogError(e, $"Failed to write ${filePath} to S3: {e.Message}");
-                throw e;
-            }
+                CallerReference = DateTime.Now.Ticks.ToString(),
+                Paths = new Paths()
+                {
+                    Items = paths,
+                    Quantity = paths.Count()
+                }
+            };
+
+            var req = new CreateInvalidationRequest(_cloudFrontConfig.DistributionID, invalidationBatch);
+            var invalidateCacheResponseCode = _cloudFrontService.CreateInvalidationAsync(req).Result.HttpStatusCode;
+
+            if (invalidateCacheResponseCode != HttpStatusCode.Created)
+                _logger.LogError($"Cache invalidation failed and resulted in a status code of {invalidateCacheResponseCode}");
```

Comment:
This would be useful to log the response id, the invalidation caller reference, the paths for debugging purposes if this call fails - so we can see info in Kibana

Reply:
And the invalidation id (CreateInvalidationResponse.Invalidation.Id - that's what you can use in the AWS GUI to find a specific invalidation)